docs/features/role-based-access-control/high-level-design.md

```diff
@@ -0,0 +1,100 @@
+# Role base access control - High-level design
+
+This document aims at providing an overview of the architecture of Role-based Access Control (RBAC from here on) for
+the Marxan platform.
+
+## Roles
+
+Platform administrators should be identified by a boolean attribute in the users table (`is_admin`). This attribute can
+only be edited by other platform administrators, and it must not be editable by the user when editing their profile.
+
+The remaining roles supported by the platform (`Owner`, `Contributor` and `Viewer`) will be stored in the `roles` 
+database in the `marxan-api` database.
```

Comment:
Requirements state more than 3 roles:

* Platform admin
* Project owner
* Project contributor
* Project reviewer (read-only)
* Project solution-viewer  [nice-to-have]

Reply:
My idea is to have generic/reusable roles, that not only cover the needs of the project but also that can be accommodate future needs - the roles described in the current requirements would be covered as such:

- "Platform admin" would be achieved with the new DB table for platform admins described here https://github.com/Vizzuality/marxan-cloud/pull/621#discussion_r744485106
- "Project owner" -> role Owner for resource Project
- "Project contributor" -> role Contributor for resource Project
- "Project reviewer (read-only)" -> role Viewer for resource Project
- "Project solution-viewer [nice-to-have]" -> role Viewer for resource "Scenarios" or "Solutions"

Do you see anything missing?

Reply:
@ikas as per https://github.com/Vizzuality/marxan-cloud/pull/621#discussion_r744587371 - "viewer" may have different aspects enabled/disabled , so I am afraid at some point we won't be able to refrain to have multiple "project/scenario viewer" roles.

Reply:
@kgajowy during the final PR review, can you check if this still applies? I think with the latest changes to the implementation plan this should not be a problem anymore, but I'd appreciate if you could double check :+1: